File: src/main/java/net/atos/client/bag/util/TypeAdresseerbaarObjectEnumAdapter.java
/*
 * SPDX-FileCopyrightText: 2023 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.client.bag.util;

import jakarta.json.bind.adapter.JsonbAdapter;

import nl.info.client.bag.model.generated.TypeAdresseerbaarObject;

public class TypeAdresseerbaarObjectEnumAdapter implements JsonbAdapter<TypeAdresseerbaarObject, String> {

    @Override
    public String adaptToJson(final TypeAdresseerbaarObject typeAdresseerbaarObject) {
        return typeAdresseerbaarObject.toString();
    }

    @Override
    public TypeAdresseerbaarObject adaptFromJson(final String json) {
        return TypeAdresseerbaarObject.fromValue(json);
    }
}

File: src/main/kotlin/nl/info/client/bag/util/TypeAdresseerbaarObjectEnumAdapter.kt
/*
 * SPDX-FileCopyrightText: 2023 Atos, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.bag.util

import jakarta.json.bind.adapter.JsonbAdapter
import nl.info.client.bag.model.generated.TypeAdresseerbaarObject

class TypeAdresseerbaarObjectEnumAdapter : JsonbAdapter<TypeAdresseerbaarObject, String> {
    override fun adaptToJson(typeAdresseerbaarObject: TypeAdresseerbaarObject): String = typeAdresseerbaarObject.toString()
    override fun adaptFromJson(json: String): TypeAdresseerbaarObject = TypeAdresseerbaarObject.fromValue(json)
}
